realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1525,6 +1525,15 @@ public void executeTransaction(Transaction transaction) {
             throw new IllegalArgumentException("Transaction should not be null");
         }
 
+        // Warn on transaction being executed on UI thread if allowWritesOnUiThread is set to true, throw otherwise
+        if (sharedRealm.capabilities.isMainThread()) {
+            if (getConfiguration().isAllowWritesOnUiThread()) {
+                RealmLog.warn("It is not recommended to run transactions on the UI thread as it may lead to a drop of frames or ANRs. Please consider using 'executeTransactionAsync' instead.");
+            } else {
+                throw new RealmException("Running transactions on the UI thread is disabled by default. You can opt in by using 'RealmConfiguration.allowWritesOnUiThread'.");
```

Comment:
This is not documented in `@throws`. The documentation could also point to `executeTransactionAsync` to guide users around this.